### PR TITLE
Implementation of a SimpleBuildWrapper for use in workflows. Change i…

### PR DESCRIPTION
**Pull request by qvicksilver**
…n parent version due to "Build wrappers" headline here: https://github.com/jenkinsci/workflow-plugin/blob/cd9b86e225c772084c39ebb110275bf076f010ad/COMPATIBILITY.md

### CODE REVIEW

**Comment by dblock** [comment]
This is great. I would merge with tests, please!

**Comment by qvicksilver** [comment]
I read about that but noticed there was no test for you build wrapper so I thought I might get away with it ;) But I'll produce some test and update the PR.

**Comment by dblock** [comment]
Well now you also have to write a test for the build wrapper :)

**Comment by omehegan** [comment]
Can we get this merged? I am itching to have this support as I convert projects to Workflow. I love AnsiColor!

File: pom.xml [hunk 1]
```diff
@@ -89,6 +89,11 @@
       <artifactId>jansi</artifactId>
       <version>1.9</version>
     </dependency>
+    <dependency>
+      <groupId>org.jenkins-ci.plugins.workflow</groupId>
+      <artifactId>workflow-aggregator</artifactId>
+      <version>1.8</version>
```
Comment (jglick) [inline pom.xml]: This should be a `test`-scoped dependency. (Or deleted, if you do not plan to add an actual `JenkinsRule`-based test verifying Workflow compatibility.)

File: src/main/java/hudson/plugins/ansicolor/AnsiColorSimpleBuildWrapper.java [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+import java.io.Serializable;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import jenkins.tasks.SimpleBuildWrapper;
+import org.apache.commons.lang.builder.HashCodeBuilder;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+public class AnsiColorSimpleBuildWrapper extends SimpleBuildWrapper implements Serializable {
```
Comment (jglick) [inline src/main/java/hudson/plugins/ansicolor/AnsiColorSimpleBuildWrapper.java]: It suffices to just change the superclass of the existing `AnsiColorBuildWrapper`. (A `SimpleBuildWrapper` can be used by traditional projects too.)

File: src/main/java/hudson/plugins/ansicolor/AnsiColorSimpleBuildWrapper.java [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+    
+    private static final long serialVersionUID = 1L;
```
Comment (jglick) [inline src/main/java/hudson/plugins/ansicolor/AnsiColorSimpleBuildWrapper.java]: Why is this being made `Serializable`? It should never be serialized AFAIK.

**Comment by dblock** [comment]
Sorry, I didn't get a notification from Github from the update - thanks for writing the tests. I think some questions from @jglick seem legit, care to take a look please @qvicksilver. If you don't mind squashing commits too.

File: pom.xml [hunk 4]
```diff
@@ -5,7 +5,7 @@
   <parent>
     <groupId>org.jenkins-ci.plugins</groupId>
     <artifactId>plugin</artifactId>
-    <version>1.424</version>
+    <version>1.609.1</version>
```
Comment (dblock) [inline pom.xml]: Is this now requiring a much newer version of Jenkins? I don't think that's what we want unless there's a strong reason.

File: pom.xml (hunk 1 above)
Comment (dblock) [inline pom.xml]: Thanks for spotting this.

File: src/main/java/hudson/plugins/ansicolor/AnsiColorSimpleBuildWrapper.java (hunk 2 above)
Comment (dblock) [inline src/main/java/hudson/plugins/ansicolor/AnsiColorSimpleBuildWrapper.java]: Yep, would be a lot less copy-paste.

File: pom.xml (hunk 4 above)
Comment (qvicksilver) [inline pom.xml]: Agreed the comments by @jglick do make sense and I'll do something about them as soon as I get the time.

The "Plugin Compatibility with Workflow"-documentation (https://github.com/jenkinsci/workflow-plugin/blob/master/COMPATIBILITY.md#build-wrappers-1) states that "To add support for a BuildWrapper, depend on Jenkins 1.599+ (typically 1.609.1), and implement SimpleBuildWrapper, following the guidelines in its Javadoc." so thats the reason for depending on 1.609.1. I guess it is a fairly strong reason.

Comment (dblock) [inline pom.xml]: :+1:

File: pom.xml (hunk 1 above)
Comment (qvicksilver) [inline pom.xml]: Changed scope and implemented a JenkinsRule based test in last commit.

**Comment by dblock** [comment]
Merging.

**Comment by jglick** [comment]
Retroactively filed as [JENKINS-30528](https://issues.jenkins-ci.org/browse/JENKINS-30528).

**Pull request merged**